sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlockBlobItem.java

```diff
@@ -31,7 +31,7 @@ public class BlockBlobItem {
      * @param encryptionKeySha256 The encryption key used to encrypt the block blob.
      */
     public BlockBlobItem(final String eTag, final OffsetDateTime lastModified, final byte[] contentMd5,
-        final boolean isServerEncrypted, final String encryptionKeySha256) {
+        final Boolean isServerEncrypted, final String encryptionKeySha256) {
```

Comment:
This breaks the public API, instead my recommendation would be changing the code calling into this constructor to convert null to false (or true if that is the default for the service)

Reply:
I recommend new constructor with `Boolean` and deprecating old constructors. Since the value describes the state of the data as the FE tells us, I would not want to make assumptions as to the state of the data if the FE has failed to tell us for whatever reason. Since the getter returns `Boolean` already, this should be safe that it sometimes is null instead of an explicit value, as we are meant to communicate the server response.